project/src/main/java/com/google/sps/data/DataAccessor.java

```diff
@@ -47,30 +56,76 @@ public DataAccessor() {
   }
 
   /**
-  * @param userId the id of the user that we want to check the registration status about
-  * @return whether the user is registered to our system, by checking whether their id was
+  * @param userId the ID of the user that we want to check the registration status for
+  * @return whether the user is registered to our system, by checking whether their ID was
   *     previously added to datastore.
   */
   public boolean isRegistered(String userId) {
-    checkArgument(!isNullOrEmpty(userId), "Invalid user ID.");
+    checkArgument(!isNullOrEmpty(userId), INVALID_USER_MSG);
     Key userIdKey = KeyFactory.createKey(USER_ENTITY_NAME, userId);
     Filter userIdFilter =
         new Query.FilterPredicate(Entity.KEY_RESERVED_PROPERTY, FilterOperator.EQUAL, userIdKey);
     Query query = new Query(USER_ENTITY_NAME).setFilter(userIdFilter).setKeysOnly();
-    return datastoreService.prepare(query)
+    return !datastoreService.prepare(query)
         .asList(FetchOptions.Builder.withDefaults())
-        .size() > 0;
+        .isEmpty();
   }
 
   /**
   * Registers the user in our system, by adding an entity that represents them to datastore.
   *
-  * @param userId the id of the user that we want to register to our system
+  * @param userId the ID of the user that we want to register to our system
   */
   public void registerUser(String userId) {
-    checkArgument(!isNullOrEmpty(userId), "Invalid user ID.");
+    checkArgument(!isNullOrEmpty(userId), INVALID_USER_MSG);
     checkArgument(!isRegistered(userId), "User already registered.");
     Entity userEntity = new Entity(USER_ENTITY_NAME, userId);
     datastoreService.put(userEntity);
   }
+
+  /**
+  * Updates the database with information from the user feedback.
+  *
+  * @param feedback a UserFeedback that holds all the information about the user feedback.
+  */
+  public void updateUserFeedback(UserFeedback feedback) {
+    for (String place : feedback.recommendedPlaces()) {
+      Entity recommendationEntity = new Entity(RECOMMENDATION_ENTITY_KIND);
+      recommendationEntity.setProperty(USER_ID_PROPERTY, feedback.userId());
+      recommendationEntity.setProperty(PLACE_ID_PROPERTY, place);
+      recommendationEntity.setProperty(CHOSEN_PROPERTY,
+          feedback.chosenPlace().isPresent() && feedback.chosenPlace().get().equals(place));
+      recommendationEntity.setProperty(TRY_AGAIN_PROPERTY, feedback.userTriedAgain());
+      recommendationEntity.setProperty(TIME_PROPERTY, feedback.feedbackTimeInMillis());
+      datastoreService.put(recommendationEntity);
+    }
+  }
+
+  /**
+  * Approches the database and gets the places that were recommended to the user in the past.
+  * It is possible to get only places that the user chose, according to their feedback.
```

Comment:
But what if `getOnlyPlacesUserChose` is false?

Reply:
I think it's clear from the rest of the documentation, you don't think so?

Reply:
Oh now that I read it again I understood what you meant in this sentence. It sounds like the only possibilty is to get places the user chose, so it didn't go along with the rest of the documentation. Maybe "there is apossibility to get.." will be more clear.

Reply:
np done